### PR TITLE
Update user functionality.

### PR DESCRIPTION
**Pull request by davidtruong**
Updates the user fields

### CODE REVIEW

File: Iterable-iOS-SDK/IterableAPI.m [hunk 1]
```diff
@@ -650,6 +650,32 @@ - (void)disableDeviceForAllUsersWithOnSuccess:(OnSuccessHandler)onSuccess onFail
...
+        } else {
+            args = @{
+                     ITBL_KEY_USER_ID: self.userId,
+                     ITBL_KEY_DATA_FIELDS: dataFields,
+                     ITBL_KEY_MERGE_NESTED: mergeNested
+                     };
+        }
+    }
+    
+    NSURLRequest *request = [self createRequestForAction:ENDPOINT_UPDATE_USER withArgs:args];
```
Comment (dontgitit) [inline Iterable-iOS-SDK/IterableAPI.m]: is this fine if `args` is null?

File: Iterable-iOS-SDK/IterableAPI.m [hunk 2]
```diff
@@ -650,6 +650,32 @@ - (void)disableDeviceForAllUsersWithOnSuccess:(OnSuccessHandler)onSuccess onFail
 }
 
 // documented in IterableAPI.h
+- (void)updateUser:(NSDictionary *)dataFields mergeNestedObjects:(BOOL)mergeNestedObjects onSuccess:(OnSuccessHandler)onSuccess onFailure:(OnFailureHandler)onFailure
+{
+    NSDictionary *args;
+    if (dataFields) {
+        NSNumber *mergeNested = [NSNumber numberWithBool:mergeNestedObjects];
```
Comment (dontgitit) [inline Iterable-iOS-SDK/IterableAPI.m:657]: why is this a number?

Comment (davidtruong) [reply Iterable-iOS-SDK/IterableAPI.m:657]: There was a coercion error when trying to assign a BOOL in a collection: Collection element of type 'BOOL' is not an objective-C object. 
https://useyourloaf.com/blog/adding-booleans-to-objective-c-arrays-and-dictionaries/

If there an alternative you can think of let me know.

File: Iterable-iOS-SDK/IterableAPI.m (hunk 1 above)
Comment (davidtruong) [reply Iterable-iOS-SDK/IterableAPI.m]: done

**Review by dontgitit — APPROVED** [review]

**Pull request merged**